src/client/opamSolution.ml

```diff
@@ -824,6 +825,90 @@ let run_hook_job t name ?(local=[]) w =
   | None ->
     Done true
 
+let rec check_and_install_depexts t packages =
+  let sys_packages =
+    if OpamStateConfig.(!r.depext_ignore_opam) then
+      OpamSwitchState.system_packages t.switch_config packages
+        ~depexts:(OpamSwitchState.depexts t)
+    else t.sys_packages
+  in
+  let avail, nf =
+    let add pkg sys_set set =
+      if not (OpamSysPkg.Set.is_empty sys_set) then
+        OpamPackage.Map.add pkg
+          (OpamSysPkg.Set.elements sys_set) set
+      else set
+    in
+    OpamPackage.Map.fold (fun pkg sys (avail,nf) ->
+        if OpamPackage.Set.mem pkg packages then
+          add pkg sys.OpamSysPkg.s_available avail,
+          add pkg sys.OpamSysPkg.s_not_found nf
+        else
+          avail,nf)
+      sys_packages (OpamPackage.Map.empty, OpamPackage.Map.empty)
+  in
+  let print_dep pkgs msg =
+    let lst =
+      OpamPackage.Map.bindings
+        (OpamPackage.Map.map (List.map OpamSysPkg.to_string) pkgs)
+    in
+    if lst = [] then "" else
+      OpamStd.Format.itemize (fun lst ->
+          Printf.sprintf "%s\n%s"
+            msg
+            (OpamStd.Format.itemize (fun (p,s) ->
+                 Printf.sprintf "%s: %s"
+                   (OpamPackage.to_string p)
+                   (OpamStd.Format.pretty_list s))
+                lst)
+        ) [lst]
+  in
+  if not (OpamPackage.Map.is_empty avail && OpamPackage.Map.is_empty nf) then
+    (OpamConsole.error "Some external dependencies are missing:\n%s%s"
+       (print_dep avail "available to install:")
```

Comment:
We print all packge-syspackage that are not installed (and required), but split given their status of `available || not found`

Reply:
That's what I thought, but on my limited tests I got `libssl-dev` listed, while it was installed :thinking:

Reply:
hmm... maybe your in the case of virtual packages. I disabled virtual package check because it takes ~20s to check all `not-found`... can you check if `libssl-dev` is returned as installed by `dpkg-query -l libssl-dev`

Reply:
```
% dpkg-query -l libssl-dev
Souhait=inconnU/Installé/suppRimé/Purgé/H=à garder
| État=Non/Installé/fichier-Config/dépaqUeté/échec-conFig/H=semi-installé/W=attend-traitement-déclenchements
|/ Err?=(aucune)/besoin Réinstallation (État,Err: majuscule=mauvais)
||/ Nom                        Version            Architecture       Description
+++-==========================-==================-==================-==========================================================
ii  libssl-dev:amd64           1.1.0j-1~deb9u1    amd64              Secure Sockets Layer toolkit - development files
```
:thinking:

Reply:
Oh, it's because the `:amd64` suffix.